### DISCUSSION TITLE
Are there PROs and CONs to the new custom tokens (SandboxieLogon)

### DISCUSSION THREAD

**Discussion by bastik-1001** [sandboxie-plus/Sandboxie#2064] [Documents and notes]
Are there PROs and CONs to the different tokens compared to the new feature and is it affected by being experimental?
> Sandboxie to use custom logon tokens instead of the anonymous

The documentation and what I have been told before, on here, is that the anonymous token, makes Windows reject calls made from such a process. Is that still the case with the custom tokens?

It seems that custom tokens provide isolation against different sandboxes, if I understand that correctly.

**Comment by isaak654** [comment]
The Sandboxie feature "SandboxieLogon" can be located on the Sandboxie-Plus UI in Options menu -> Global Settings -> Advanced Config:
![SandboxieLogon](https://user-images.githubusercontent.com/12372772/181879229-3ce5abbf-9ed6-41b4-a082-1776e26cdb86.png)

> It seems that custom tokens provide isolation against different sandboxes, if I understand that correctly.

Indeed, that's what it does:
<img src="https://user-images.githubusercontent.com/12372772/181503242-6c3b3224-0dd4-4f6d-952b-ba2c9749a170.png" alt="SID_per_sandbox" height="350px">
Source (in case the picture disappears): https://www.wilderssecurity.com/threads/sandboxie-plus-v1-2-0.445783/#post-3090427

There is a known limitation about `SandboxieLogon=y`, which is applied internally by default from v1.2.0 / 5.57.0 to v1.2.6 / 5.57.6:
>  - SSL connection fails with 3rd party anti malware tools when SSL scanning is enabled, the fix is to set `SandboxieLogon=n`
>  https://github.com/sandboxie-plus/Sandboxie/issues/2025#issuecomment-1196343325

Moral of the story: if no one reports the issue (with the most accurate details) in the AVs support forums, it could become a wasted feature for everyone but informed Sandboxie users.

**Reply by bastik-1001** [reply]
Provided that SandboxieLogon=y does not have any compatibly issues, is it as secure as the anonymous login?

Reading the changelog for the upcoming 1.2.7 I noticed that the new feature had some downside, since the version disables it. Considering the implications of the issue, it is justified, but sad nonetheless.

Their SSL/TLS inspection is a double-edged sword. Obviously, users trust their AV vendors and let it mess with their connections. I disliked the practice of them acting that way and would opt to disable it, but others might not want to do that, even feeling that it would be a tradeoff, if they disable something, they find acceptable.

Since it is a compatibility issue, maybe sandman.exe could check for the presence of an AV vendor's product that is known to conflict and set SandboxieLogon=n. It may be also part of the configuration wizard. I have the feeling that someone installing sandboxie, knows what kind of AV software they are using, so they could enable/disable it. Sure, in an ideal world that would not be needed and the vendors apply the required changes.

**Reply by diversenok** [reply]
> Provided that SandboxieLogon=y does not have any compatibly issues, is it as secure as the anonymous login?

It is more secure. The primary purpose of using custom SIDs is to isolate sandboxes from each other and prevent cross-sandbox attacks. Using the same user (ANONYMOUS LOGON) for sandboxes with different levels of permissions allows programs from hardened sandboxes to inject code into processes from other more permissive sandboxes. Using individual virtual accounts for each sandbox solves this problem. Secondly, ANONYMOUS LOGON is a built-in SID, so you might theoretically find resources that explicitly grant access to it, although unlikely. Sandboxie SIDs are custom, which eliminates this possibility.

**Reply by bastik-1001** [reply]
Thank you for explaining that so nicely.

**Reply by bastik-1001** [reply]
It's useful to know that SandboxieLogon=y can be applied per box, as well. This makes it possible to use that feature for some boxes, even with conflicting AV software installed, when those products only mess with browsers.

**Reply by joy-maruyama** [reply]
bastik-1001, are you sure that SandboxieLogon can be applied per box ? That would be very good news!

**Reply by bastik-1001** [reply]
Joy-1101, I have not tested it myself, but David himself [stated that](https://github.com/sandboxie-plus/Sandboxie/discussions/2075#discussioncomment-3279281) in another discussion/documentation/support question.

> The SandboxieLogon should be set glboally but can be set per box (...)

Edit: Now I tested it with version 1.2.7, SandboxieLogon=y is globally set (manually) and gets applied to boxes; they get different SIDs. Then I added SandboxieLogon=n to one of the boxes and ProcessExplorer showed that it used the built-in anonymous SID.

**Reply by DavidXanatos** [reply]
yes it can, 
you can even do the following
add  SandboxieLogon=y globally
and  SandboxieLogon=n to he box you run your browser in, 
and et voila boxes are isolated and browser still works with ssl scanning

**Reply by joy-maruyama** [reply]
That is amazing! Thank you very much!

**Reply by joewiey** [reply]
Support and update
